presto-main/src/main/java/com/facebook/presto/GroupByHashPageIndexer.java

```diff
@@ -51,7 +53,10 @@ public GroupByHashPageIndexer(GroupByHash hash)
     @Override
     public int[] indexPage(Page page)
     {
-        GroupByIdBlock groupIds = hash.getGroupIds(page);
+        Work<GroupByIdBlock> work = hash.getGroupIds(page);
+        // TODO: this class does not yield wrt memory limit; enable it
+        verify(work.process());
```

Comment:
It should be clear who is responsible for calling `process`. Is it the one who creates the work or the one using it? Since you're going to add yielding i guess you will store the work on a field, then you'll call process here. Thus let the one who creates work not call `process`

Reply:
I was hesitating choosing between these two. Your point makes sense. Will move this outside to the caller.